src/TreeSupportTipGenerator.cpp

```diff
@@ -1198,8 +1198,18 @@ void TreeSupportTipGenerator::generateTips(
         {
             if (layer_idx > 0)
             {
-                storage.support.supportLayers[layer_idx].support_fractional_roof.add(
-                    storage.support.supportLayers[layer_idx].support_roof.difference(storage.support.supportLayers[layer_idx + 1].support_roof));
+                // Difference between support roofs of adjacent layers
+                auto paths = storage.support.supportLayers[layer_idx].support_roof.difference(storage.support.supportLayers[layer_idx + 1].support_roof);
+                // Current support fractional roof
+                auto& path_in_sfr = storage.support.supportLayers[layer_idx].support_fractional_roof;
+                // Loop over paths and add to fractional roof if not already present
+                for (const auto& path : paths)
+                {
+                    if (std::find(path_in_sfr.begin(), path_in_sfr.end(), path) == path_in_sfr.end())
+                    {
+                        path_in_sfr.add(path);
+                    }
+                }
```

Comment:
Please use `auto` only when the type is really explicit or very complex (X3)

Reply:
https://github.com/Ultimaker/CuraEngine/pull/2052/commits/585102595a01e81082396253783fc996ef63ce6f